super32emu/super32emu/logic/emulator.py

```diff
@@ -165,7 +170,15 @@ def __arithmetic_instruction(self, first_source: str, second_source: str, target
         elif func == self.commands['arithmetic']['OR']:
             result = r1_value | r2_value
         elif func == self.commands['arithmetic']['NOR']:
-            result = r1_value | r2_value ^ 0b11111111
+            result = r1_value | r2_value ^ 0xffffffff
+        elif func == self.commands['arithmetic']['NAND']:
+            result = r1_value & r2_value ^ 0xffffffff
+        elif func == self.commands['arithmetic']['SHL']:
+            result = r1_value << r2_value
+        elif func == self.commands['arithmetic']['SLR']:
+            result = r1_value >> r2_value
+        elif func == self.commands['arithmetic']['SAR']:
+            result = (r1_value >> r2_value) + (r1_value >> 31) * (0xffffffff >> (32 - r2_value) << (32 - r2_value))
```

Comment:
I hope this line works as expected :sweat_smile:

Reply:
Me too 😂